### PR TITLE
Create test plan for root_group extension

### PR DESCRIPTION
**Pull request by kopylovanat**
Create test plan for extension [sycl_ext_oneapi_root_group](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_root_group.asciidoc).

### CODE REVIEW

File: test_plans/root_group.asciidoc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+auto bundle = sycl::get_kernel_bundle(q.get_context());
+auto kernel = bundle.get_kernel<class KernelName>();
+auto maxWGs = kernel.ext_oneapi_get_info<sycl::ext::oneapi::experimental::info::kernel_queue_specific::max_num_work_group_sync>(q);
+auto nd_range = sycl::nd_range<Dimensions>{global_range, local_range};
+auto props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
+q.parallel_for<class KernelName>(nd_range, props, [=](sycl::nd_item<Dimensions> it) {
+    auto root = it.ext_oneapi_get_root_group();
+});
+----
+where `global_range.size()` = `local_range.size() * maxWGs`
```
Comment (gmlueck) [inline test_plans/root_group.asciidoc:37]: There should be a test someplace that `maxWGs` is at least `1`.

File: test_plans/root_group.asciidoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  SKIP("maxWGs should be at least 1");
+auto nd_range = sycl::nd_range<Dimensions>{global_range, local_range};
+auto props = sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::use_root_sync};
+q.parallel_for<class KernelName>(nd_range, props, [=](sycl::nd_item<Dimensions> it) {
+    auto root = it.ext_oneapi_get_root_group();
+});
+----
+where `global_range.size()` = `local_range.size() * maxWGs`.
+
+`maxWGs` should be at least `1`, otherwise test for this `Dimensions` is skipped.
```
Comment (gmlueck) [inline test_plans/root_group.asciidoc:41]: Sorry, I meant that the test should fail if `maxWGs` is less than `1`.  The spec says that the value returned by `kernel_queue_specific::max_num_work_group_sync` must be at least `1`.

Comment (gmlueck) [reply test_plans/root_group.asciidoc]: Up to you.  I think this is fine.

**Review by gmlueck — APPROVED** [review]
I have no more comments, so marking this "approved".

However, we shouldn't start writing tests for this extension until the extension is finalized.  Do we want to keep this PR pending until then, or merge it with a note saying that the extension is not yet finalized?

**Comment by keryell** [comment]
I have converted this to a draft while you think about it.

**Comment by fraggamuffin** [comment]
this is for an vendor extension, so its an exception to the 1company rule.
If we remove the draft, we will get more reviews.

**Comment by gmlueck** [comment]
> @gmlueck, @keryell, could either of you mark this PR as ready for review? This test plan is up-to-date with the extension and we don't anticipate any further changes.

I marked it as ready for review.

**Comment by tomdeakin** [comment]
Ready to merge.

**Pull request merged**